msal/src/internal/java/com/microsoft/identity/client/MsalUtils.java

```diff
@@ -96,6 +110,54 @@ static boolean isEmpty(final String message) {
         return message == null || message.trim().length() == 0;
     }
 
+    /**
+     * Get the connect timeout.
+     *
+     * @return connect timeout
+     */
+    public int getConnectTimeOut() {
+        return mConnectTimeOut;
+    }
+
+    /**
+     * Sets the maximum time in milliseconds to wait while connecting.
+     * Connecting to a server will fail with a SocketTimeoutException if the
+     * timeout elapses before a connection is established. Default value is
+     * 30000 milliseconds.
+     *
+     * @param timeOutMillis the non-negative connect timeout in milliseconds.
+     */
+    public void setConnectTimeOut(int timeOutMillis) {
+        if (timeOutMillis < 0) {
+            throw new IllegalArgumentException("Invalid timeOutMillis");
+        }
+        this.mConnectTimeOut = timeOutMillis;
+    }
+
+    /**
+     * Get the read timeout.
+     *
+     * @return read timeout
+     */
+    public int getReadTimeOut() {
+        return mReadTimeOut;
+    }
+
+    /**
+     * Sets the maximum time to wait for an input stream read to complete before
+     * giving up. Reading will fail with a SocketTimeoutException if the timeout
+     * elapses before data becomes available. The default value is 30000.
```

Comment:
Regarding long-term ease of maintenance, I would consider making `DEFAULT_READ_TIMEOUT` `public final` and amending the last sentence of this javadoc to say "The default value is `DEFAULT_READ_TIMEOUT`".

I would consider the same approach for `DEFAULT_CONNECT_TIMEOUT`.

This way, if the value changes in the future, it needs only to be updated in a single spot and the documentation is always correct.

Reply:
If you do plan to make `DEFAULT_READ_TIMEOUT` and `DEFAULT_CONNECT_TIMEOUT` `public`, I would also suggest adding inline javadoc for those fields.